encoders/numeric/CompressionVaeEncoder/__init__.py

```diff
@@ -16,68 +18,63 @@ class CompressionVaeEncoder(BaseNumericEncoder, BaseTFEncoder):
-        self.to_device()
+        from cvae import cvae
+        import cvae.lib.model_iaf as model
+        import tensorflow as tf
+
+        params_path = os.path.join(self.model_path, 'params.json')
+
+        config = tf.ConfigProto(log_device_placement=False)
+        config.gpu_options.allow_growth = True
+
+        with tf.Graph().as_default():
+            if os.path.exists(params_path):
+                # Load parameter file.
+                with open(params_path, 'r') as f:
+                    param = json.load(f)
+
+                net = model.VAEModel(param,
+                                     None,
+                                     input_dim=param['dim_feature'],
+                                     keep_prob=tf.placeholder_with_default(input=tf.cast(1.0, dtype=tf.float32),
+                                                                           shape=(),
+                                                                           name="KeepProb"),
+                                     initializer='orthogonal')
+                # Placeholder for data features
+                self.data_feature_placeholder = tf.placeholder_with_default(
+                    input=tf.zeros([64, param['dim_feature']], dtype=tf.float32),
+                    shape=[None, param['dim_feature']])
 
-    @batching
+                self.embeddings = net.embed(self.data_feature_placeholder)
+
+                self.sess = tf.Session(config=config)
+                init = tf.global_variables_initializer()
+                self.sess.run(init)
+
+                # Saver for loading checkpoints of the model.
+                saver = tf.train.Saver(var_list=tf.trainable_variables())
+                cvae.load(saver, self.sess, self.model_path)
+
+                self.to_device()
+
+    @batching()
     @as_ndarray
     def encode(self, data: 'np.ndarray', *args, **kwargs) -> 'np.ndarray':
         """
         :param data: a `B x T` numpy ``ndarray``, `B` is the size of the batch
         :return: a `B x D` numpy ``ndarray``
         """
-        from cvae import cvae
-        model = cvae.CompressionVAE(X=self.X,
-                                    train_valid_split=self.train_valid_split,
-                                    dim_latent=self.output_dim,
-                                    iaf_flow_length=self.iaf_flow_length,
-                                    cells_encoder=self.cells_encoder,
-                                    initializer=self.initializer,
-                                    batch_size=self.batch_size,
-                                    batch_size_test=self.batch_size_test,
-                                    logdir=self.model_path,
-                                    feature_normalization=self.feature_normalization,
-                                    tb_logging=self.tb_logging)
-        return model.embed(data)
+        import tensorflow as tf
+
+        return self.sess.run([self.embeddings],
```

Comment:
Just cueious, what are the return values of session run? Why is the first index the one we want in return?

Reply:
It returns a ndarray with shape (1 x batch size x number of features of embedding) where 1 means one batch.